METAPROGRAMMING.md

```diff
@@ -1,58 +1,67 @@
-- Decorators can manipulate private fields and private methods, whereas operations on objects are unable to manipulate the definitions of private names.
-- Decorators do their manipulations at the time the class is defined, rather than when instances are being created and used, so they may lead to patterns which are more amenable to efficient implementation.
 
-Decorators are implemented as functions which take a JSON-like representation of a class element as an argument and return a possibly-modified form of that, optionally with additional class elements. Class elements are represented in the form: 
+* Decorators make it more clear what's going on--users write class literals and explicitly annotate the piece of source code that is being manipulated, rather than spooky action-at-a-distance where a big object is passed into a special application-dependent class framework.
+* Decorators can manipulate private fields and private methods, whereas operations on objects are unable to manipulate the definitions of private names.
+* Decorators do their manipulations at the time the class is defined, rather than when instances are being created and used, so they may lead to patterns which are more amenable to efficient implementation.
+
+As stated in the readme, decorators can be used either to decorate a whole class or an individual class element (field or method). Decorators are implemented as functions which take a JSON-like representation of a class element as an argument and return a possibly-modified form of that, optionally with additional class elements. For both kinds of decorator functions, class elements are represented in the form:
 
 ```js
 {
   kind: "method" or "field"
   key: String, Symbol or Private Name,
   placement: "static", "prototype" or "own",
   descriptor: Property Descriptor (argument to Object.defineProperty),
+  initializer: method used to set the initial state of the class element
 }
 ```
 
-The return value is allowed to add an `extras` property with additional class elements, and a `finisher` property, which is a callback that is called with the constructor, once it's created.
+For class element decorators, the return value is allowed to add the following additional properties:
+
+* `extras`: A property with additional class elements
+* `finisher`: A callback that is called with the constructor, once it's created.
 
-Class decorators are passed in an Array of all class elements and output an object with fields `elements` for the new class elements, `constructor` for the function which should act as the construtor, and `finisher` for a similar callback.
+Class decorators (i.e. decorators that decorate the class as a whole) are passed in an Array of all class elements and output an object with fields `elements` for the new class elements, `constructor` for the function which should act as the construtor, and `finisher` for a similar callback.
```

Comment:
Seems like you're trying to clarify the wording here. What if you did this clarification by giving an example of a usage of a class decorator, method decorator and field decorator, and what the descriptors for those usages would look like?